Library/MapleBacon/MapleBacon/ImageDownloadOperation.swift

```diff
@@ -7,113 +7,117 @@ import UIKit
+                delegateQueue: OperationQueue.main())
         resumeDownload()
     }
 
     public override func cancel() {
-        task?.cancelByProducingResumeData { [unowned self] data in
+        task?.cancel(byProducingResumeData: {
+            (data) in
+            
             self.resumeData = data
-            self.finished = true
-        }
+            self._finished = true
+        })
     }
 
     private func resumeDownload() {
-        let newTask: NSURLSessionDownloadTask?
+        let newTask: URLSessionDownloadTask?
         if let resumeData = resumeData {
-            newTask = session?.downloadTaskWithResumeData(resumeData)
+            newTask = session?.downloadTask(withResumeData: resumeData)
         } else {
-            newTask = session?.downloadTaskWithURL(imageURL)
+            newTask = session?.downloadTask(with: imageURL)
         }
         newTask?.resume()
         task = newTask
     }
-
-    private var _finished = false
-    override public var finished: Bool {
-        get {
-            return _finished
-        }
-        set {
-            willChangeValueForKey("isFinished")
-            _finished = newValue
-            didChangeValueForKey("isFinished")
-        }
-    }
-
 }
 
-extension ImageDownloadOperation: NSURLSessionDownloadDelegate {
+extension ImageDownloadOperation: URLSessionDownloadDelegate {
 
-    public func URLSession(session: NSURLSession, downloadTask: NSURLSessionDownloadTask,
+    public func urlSession(_ session: Foundation.URLSession, downloadTask: URLSessionDownloadTask,
                            didWriteData bytesWritten: Int64, totalBytesWritten: Int64,
                            totalBytesExpectedToWrite: Int64) {
+        
         progress.totalUnitCount = totalBytesWritten
         progress.completedUnitCount = bytesWritten
-        delegate?.imageDownloaderDelegate(self, didReportProgress: progress)
+        delegate?.imageDownloaderDelegate(downloader: self, didReportProgress: progress)
     }
 
-    public func URLSession(session: NSURLSession, downloadTask: NSURLSessionDownloadTask,
-        didFinishDownloadingToURL location: NSURL) {
+    public func urlSession(_ session: Foundation.URLSession, downloadTask: URLSessionDownloadTask,
+                           didFinishDownloadingTo location: URL) {
         do {
-            let newData = try NSData(contentsOfURL: location, options: NSDataReadingOptions.DataReadingMappedIfSafe)
-            let newImage = UIImage.imageWithCachedData(newData)
+            
+            let newData = try Data(contentsOf: location, options: Data.ReadingOptions.dataReadingMappedIfSafe)
+            let newImage = UIImage.image(withCachedData: newData)
             let newImageInstance = ImageInstance(image: newImage, data: newData, state: .New, url: imageURL)
-            if (self.cancelled == true) { return }
+            if self.isCancelled == true { return }
```

Comment:
ugh, why was this doing `== true` in the first place. just `if self.isCancelled …`